### PR TITLE
Clears #5909

### PR DESCRIPTION
**Pull request by Adenilson**
Recent changes on Fragment::establishes_stacking_context() makes this hack no longer necessary for displaying filtered inline elements.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7302)

### CODE REVIEW

File: components/layout/display_list_builder.rs [hunk 1]
```diff
@@ -1819,15 +1819,10 @@ impl InlineFlowDisplayListBuilding for InlineFlow {
...
-        // and remove the check for filter presence. Further details on #5812.
-        has_stacking_context = has_stacking_context && {
-            if let SpecificFragmentInfo::Canvas(_) = self.fragments.fragments[0].specific {
-                true
-            } else {
-                !self.fragments.fragments[0].style().get_effects().filter.is_empty()
-            }
-        };
+        match self.fragments.fragments[0].specific {
+            SpecificFragmentInfo::Canvas(_) => has_stacking_context = has_stacking_context && true,
```
Comment (pcwalton) [inline components/layout/display_list_builder.rs]: `&& true`?

Comment (Adenilson) [inline components/layout/display_list_builder.rs]: Looks to be the case for "|| true" or simple set has_stacking_context = true;

I'm assuming that we need a stacking context when there is Canvas?

**Comment by jdm** [comment]
```

failures:
    acid1_a.html == acid1_b.html
    iframe/stacking_context.html == iframe/stacking_context_ref.html
    inline_block_stacking_context_a.html == inline_block_stacking_context_ref.html
```

**Comment by Adenilson** [comment]
I was able to reproduce the error in OSX, more info: https://gist.github.com/Adenilson/4eb3ab62247ffbc625b8

I applied a check for the Fragments vector length and it stops the panic from happening.

**Comment by Adenilson** [comment]
That solves the case for acid1, I'm investigating what is wrong with the other 2 tests.

**Comment by Adenilson** [comment]
Did some digging, the failing tests have associated SC (Stacking Context) but won't draw correctly. It seems that the original 'stacking_context && something_something' was required to turn to false the stacking_context flag.

Otherwise the element won't be displayed. It is a bit unclear if that is thanks to Paint code or past DL (Display List) optimization.

I also noticed that there is no need for checking the Canvas fragment presence, as it is already done as a special case in Fragment::establishes_stacking_context().

The current patch removes this unnecessary check and documents this issue. I executed all the 354 ref tests to make sure that there is no regression.

**Comment by Adenilson** [comment]
Anyone could kick the bots on this one?

**Comment by pcwalton** [comment]
@mrobinson Is this fixed now with your stuff?

**Comment by mrobinson** [comment]
@pcwalton I think that this requires a further fix, which is to also integrate stacking contexts into the list of inlines. I've started working on that today, though I think it will be a multi-stage process.

**Comment by frewsxcv** [comment]
What is the status of this?

**Comment by mrobinson** [comment]
I think that https://github.com/servo/servo/pull/8376 makes this PR unnecessarily. It fixes the issue that I mention in my comment. Additionally, it turns out that inlines with stacking contexts belong in the list of positioned content to match what other browsers do.

**Pull request closed without merging**